### PR TITLE
Implementing sorting in digest

### LINKED ISSUE #7390: Sorting does not work in digest
### Please describe the problem (or idea)

> What happened just before the problem occurred? Or what problem could this idea solve?
Sorting doe not work in digest
**Current Behaviour**
sorting

> What did you expect to see that you didn't?

### Please show us where to look

https://publiclab.org/ ...

### What's your PublicLab.org username?

> This can help us diagnose the issue: 

### Browser, version, and operating system

> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by sssash18**
Fixes #7390 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below
**New Behaviour**
Untitled_ Jan 30, 2020 7_29 PM

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by sssash18** [comment]
@publiclab/reviewers Actually travis is failing for reasons not related to this issue.Please help me fix it.Thanks.

**Comment by VladimirMikulic** [comment]
@sssash18 please rebase the branch and push again. `git rebase master; git push`

**Comment by sssash18** [comment]
> @sssash18 please rebase the branch and push again. `git rebase master; git push`

Thanks @VladimirMikulic .Please review the changes.

**Comment by VladimirMikulic** [comment]
@sssash18 could you resolve the issues with CodeClimate? Thanks.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -211,4 +211,22 @@ def logged_in_as(roles)
     end
     has_valid_role
   end
+
+  def sort_feed(wikis)
```
Comment (Uzay-G) [inline app/controllers/application_controller.rb]: I don't think we should put this method in the application controller. Instead we can place it in the `node` model and then call it in the view.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -211,4 +211,22 @@ def logged_in_as(roles)
...
+    elsif params[:sort] == "page_views"
+      @sorted_wikis = wikis.sort_by(&:views)
+      @sorted_wikis.reverse!
+    elsif params[:sort] == "likes"
+      @sorted_wikis = wikis.sort_by(&:cached_likes)
+      @sorted_wikis.reverse!
+    else
+      @sorted_wikis = wikis
+    end
+    @sorted_wikis
```
Comment (Uzay-G) [inline app/controllers/application_controller.rb]: You can remove this and it will return the last thing you called. So `@sorted_wikis.reverse`w

**Comment by sssash18** [comment]
@Uzay-G I think it would be better if we put it in subscriptions controller.What are your views?

**Comment by Uzay-G** [comment]
> @Uzay-G I think it would be better if we put it in subscriptions coIntroller.What are your views?

I don't think this really suits a controller method. Could you check if putting it in the node model works?

**Comment by sssash18** [comment]
@Uzay-G I don't 

> > @Uzay-G I think it would be better if we put it in subscriptions coIntroller.What are your views?
> 
> I don't think this really suits a controller method. Could you check if putting it in the node model works?

I checked it but was not able to apply a model method to a array of 'wiki' objects

File: app/controllers/application_controller.rb (hunk 2 above)
Comment (cesswairimu) [reply app/controllers/application_controller.rb]: Hey @sssash18, did you get some time to checkout @Uzay-G review above :point_up:

**Comment by cesswairimu** [comment]
Nm just saw your explanation.. I am commuting.. I will get you some pointers on how you can move that to the model. Once I get home . Thanks

**Comment by cesswairimu** [comment]
Hey @sssash18, sorry forgot to reply to this
on the node model..you can do something like
```
def self.sort_feed(wikis, sorting_type)
    case sorting_type
    when 'title'
      wikis.sort_by { |wiki| wiki.title.downcase }
    when 'edits'
      wikis.sort_by { |wiki| wiki.revisions.length }
    when 'page_views'
      wikis.sort_by(&:views)
    when 'likes'
      wikis.sort_by(&:cached_likes)
    else
      wikis
    end
  end
```
Then call it on the view like `sort_feed(wikis, params[:sort])` What do you think? Thanks

**Comment by sssash18** [comment]
@cesswairimu  Thanks for your guidance . I had opened a new PR #7481 with the required changes.Kindly review it

**Pull request closed without merging**